APKBUILD.5

```diff
@@ -0,0 +1,488 @@
+function ensures the build directories are set up correctly and applies any
+*.patch files specified in
+.Cm source .
+You must call
+.Cm default_prepare
+if you write a custom
+.Cm prepare
+function.
+
+.It Cm build
+Compiles the source in
+.Ev builddir .
+You must implement this function yourself.  If no compilation is required, you
+may simply write
+.Li return 0
+to skip the build phase.
+
+.It Cm check
+Runs the package's test suite.  This function must be implemented unless
+.Li Dq !check
+was specified in
+.Cm options .
+
+.It Cm package
+Installs the package into
+.Ev pkgdir .
+Note that
+.Ev pkgdir
+is not created for you; if this package installs no files (for example, a
+metapackage), you must use
+.Li mkdir -p "$pkgdir"
+to skip the package phase.
+
+.El
+
+
+.Ss Install Scripts
+An install script is run when an action is taken on a package by
+.Xr apk 8 .
+An install script must be written in shell and must have a
+.Li Dq #!/bin/sh
+interpreter declaration as the first line.  The
+.Cm install
+variable must contain the install scripts needed by the package.  The different
+actions available are as follows:
+
+.Bl -tag -width Ds
+.It Ic $pkgname.pre-install
+Executed before the package is installed.  If this script exits with an error
+(non-zero exit code),
+.Xr apk 8
+will halt the installation and the package will not be installed.  This install
+script is typically used to create any users or groups needed as described in
+.Cm pkggroups
+and
+.Cm pkgusers .
+
+.It Ic $pkgname.post-install
+Executed after the package is installed.  If this script exits with an error
+(non-zero exit code),
+.Xr apk 8
+will mark the package as broken.  The
+.Li apk fix
+command will attempt to re-run the post-install script if this occurs.
+
+.It Ic $pkgname.pre-upgrade
+Executed before the package is upgraded.  If this script exits with an error
+(non-zero exit code),
+.Xr apk 8
+will mark the package as broken.
+
+.It Ic $pkgname.post-upgrade
+Executed after the package is upgraded.  If this script exits with an error
+(non-zero exit code),
+.Xr apk 8
+will mark the package as broken.  The
+.Li apk fix
+command will attempt to re-run the post-upgrade script if this occurs.
+
+.It Ic $pkgname.pre-deinstall
+Executed before the package is removed from the system.  If this script exits
+with an error (non-zero exit code),
+.Xr apk 8
+will not remove the package from the system.
+
+.It Ic $pkgname.post-deinstall
+Executed after the package is removed from the system.  Exiting with an error
+will have no effect.
+
+.El
+
+
+.Sh IMPLEMENTATION NOTES
+Currently,
+.Nm
+files are sourced as normal shell scripts.  This may change at a later date.
+The
+.Xr abuild 1
+utility as distributed by Alpine uses the ash shell, a part of
+.Xr busybox 1
+that is currently undocumented.  It is mostly compliant with
+.St -p1003.2 ,
+but this cannot be fully relied upon.  The
+.Xr abuild 1
+utility as distributed by Adélie uses whatever the user has set as /bin/sh,
```

Comment:
Why do you mention Adélie in man page of **Alpine Linux's** tool?

Reply:
Similar to how the BSDs note in man pages how other BSD flavours work to ensure maximum compatibility, and how glibc notes in man pages how Solaris and kernel 2.4 works to ensure maximum compatibility, I thought it would be good to note how both distros that use `abuild` work.

I was under the impression that `abuild` was not only "Alpine" tool.  It is a community tool used by Alpine, Adélie, pmOS, etc...

I can remove this if it is desired.

Reply:
Good point, it makes sense; so let’s keep it.